build/commands/lib/test.js

```diff
@@ -216,8 +216,20 @@ const runTests = (passthroughArgs, suite, buildConfig, options) => {
       'browser_tests',
       ...getChromiumUnitTestsSuites(),
     ]
+
+    const testsToRun = getTestsToRun(config, suite)
+    // When outputting results into xml files, delete any previous results and
+    // add all expected output results file names into ${suite}.txt file.
+    if (options.output_xml) {
+      let expectedOutputs = ''
+      testsToRun.forEach((testSuite) => {
```

Comment:
I don't think this is actually necessary because the xml files are only added to the suite.txt file as they run so only that file needs to be deleted

Reply:
Couldn't it be possible that an xml file is left from a previous run and in the current run that suite fails early on and doesn't overwrite the xml file. Then CI would just parse the old xml file.